services/src/main/java/org/keycloak/services/resources/account/LinkedAccountsBean.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.services.resources.account;
+
+import org.keycloak.models.FederatedIdentityModel;
+import org.keycloak.models.IdentityProviderModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.models.utils.KeycloakModelUtils;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import org.keycloak.credential.CredentialModel;
+
+/**
+ * @author Stan Silvert
+ */
+public class LinkedAccountsBean {
+
+    private final List<LinkedAccountEntry> linkedAccounts;
+    private final boolean hasPasswordCredential;
```

Comment:
Maybe it should be renamed to "hasAlternativeAuthentication" or something more generic, considering we may add other types of login in the future. This is really there to prevent user from removing last IdP which would block them from logging in.